core/server/data/migrations/utils.js

```diff
@@ -0,0 +1,244 @@
+// @ts-check
+const ObjectId = require('bson-objectid').default;
+const logging = require('../../../shared/logging');
+
+/**
+ * Creates a migration which will add a permission to the database
+ *
+ * @param {Object} config
+ * @param {string} config.name - The name of the permission
+ * @param {string} config.action - The action_type of the permission
+ * @param {string} config.object - The object_type of the permission
+ *
+ * @returns {Migration}
+ */
+function addPermission(config) {
+    return createTransactionalMigration(
+        async function up(connection) {
+            const existingPermission = await connection('permissions').where({
+                name: config.name,
+                action_type: config.action,
+                object_type: config.object
+            }).first();
+
+            if (existingPermission) {
+                logging.warn(`Permission for ${config.action}:${config.object} already added`);
+                return;
+            }
+
+            logging.info(`Adding permission for ${config.action}:${config.object}`);
+
+            const date = connection.raw('CURRENT_TIMESTAMP');
+
+            await connection('permissions').insert({
+                id: ObjectId.generate(),
+                name: config.name,
+                action_type: config.action,
+                object_type: config.object,
+                created_at: date,
+                created_by: 1,
```

Comment:
Probably want to consider what we do about this? I think a "migration" id would be v. useful

Reply:
Yeah, I'd separate this out to a variable at least and to full on additional query to the db to check who's the owner in later refinements. The letter is probably out of scope for current task at hand, but makes it a lot more manageable to do in the future :heavy_plus_sign: :100:

Reply:
:+1: Cool, I'll pull that out to a variable!